android/app/src/main/java/com/microsoft/codepush/react/CodePushPackage.java

```diff
@@ -153,12 +153,13 @@ public WritableMap getPackage(String packageHash) {
 
     public void downloadPackage(Context applicationContext, ReadableMap updatePackage,
                                 DownloadProgressCallback progressCallback) throws IOException {
-
-        String newPackageFolderPath = getPackageFolderPath(CodePushUtils.tryGetString(updatePackage, PACKAGE_HASH_KEY));
-        if (FileUtils.fileAtPathExists(newPackageFolderPath)) {
+        String newUpdateHash = CodePushUtils.tryGetString(updatePackage, PACKAGE_HASH_KEY);
+        String newUpdateFolderPath = getPackageFolderPath(newUpdateHash);
```

Comment:
Yep. I think for ease of review I'll do it in another PR.

Some of them also involve changes in the API, e.g getCurrentPackage -> getCurrentUpdate, which may break users, so I want to leave it off till we have discussed this further internally.